src/Microsoft.AspNet.Razor/SourceLocation.cs

```diff
@@ -76,12 +128,37 @@ public static SourceLocation Advance(SourceLocation left, string text)
             return tracker.CurrentLocation;
         }
 
-        public static SourceLocation Add(SourceLocation left, SourceLocation right)
+        private static SourceLocation CreateUndefined()
         {
+            var sl = new SourceLocation();
```

Comment:
Since you're touching it just do 

```C#
var sl = new SourceLocation
{
    Absoluteindex = -1,
    LineIndex = -1,
    CharacterIndex = -1
};
```